codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Connection.java

```diff
@@ -253,6 +253,30 @@
     }
 
     /**
+     * A key to be used for associating application-defined properties with streams within this connection.
+     *
+     * @param <K> The object type underlying the key.
+     * @param <V> The value type to be stored by this key.
+     */
+    interface PropertyKey<K, V> {
```

Comment:
Thanks for clarifying.  I think associating the key/value in the generic type arguments is a bit confusing by coupling the key type to the value type.  I would prefer 1 or 3 but am open for discussion.